element-hq/element-x-android
# Require acknowledgement to send to a verified user if their identity changed or if a device is unverified.

## Content
This PR makes the following changes:

Build clients with the requirements to fail when sending to once verified users who aren't (fully) verified any more
Show a description of the send failures for verified users in the ActionListView
Add a bottom sheet to resolve failures for verified users by tapping on the description above, or by tapping on the failure in the timeline.
If there is more than one user ID in the failure, the sheet iterates through them 1 by 1, resolving each one and then dismissing itself at the end.


## Motivation and context
Closes https://github.com/element-hq/element-x-android/issues/3459
Closes https://github.com/element-hq/element-x-android/issues/3460


## Screenshots / GIFs



## Tests



- Makes sure to verify a user with another client
- Reset the security or login with a new device for this verified user
- Send a message with EX in an encrypted room with this user
- See the message failing to send 
- Click on the error icon or long click on the item
- See the bottomsheet and play with the options

## Tested devices

- OS version(s):

## Checklist
